db/migrations/9_create_subscriptions.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE public.subscriptions_set
+(
+    id                     uuid                                 DEFAULT public.gen_random_uuid() NOT NULL,
+    name                   character varying           NOT NULL,
+    state                  character varying           NOT NULL DEFAULT 'initial',
+    restarts_count         integer                     NOT NULL DEFAULT 0,
+    max_restarts_number    int2                        NOT NULL DEFAULT 10,
+    time_between_restarts  int2                        NOT NULL DEFAULT 1,
+    last_restarted_at      timestamp without time zone,
+    last_error             jsonb,
+    last_error_occurred_at timestamp without time zone,
+    created_at             timestamp without time zone NOT NULL DEFAULT now(),
+    updated_at             timestamp without time zone NOT NULL DEFAULT now()
+);
+
+ALTER TABLE ONLY public.subscriptions_set
+    ADD CONSTRAINT subscriptions_set_pkey PRIMARY KEY (id);
+
+CREATE TABLE public.subscriptions
+(
+    id                           bigserial                   NOT NULL,
+    set                          character varying           NOT NULL,
+    name                         character varying           NOT NULL,
+    options                      jsonb                       NOT NULL DEFAULT '{}'::jsonb,
+    events_processed_total       bigint                      NOT NULL DEFAULT 0,
+    current_position             bigint,
+    average_event_time           float4,
+    state                        character varying           NOT NULL DEFAULT 'initial',
+    restarts_count               integer                     NOT NULL DEFAULT 0,
+    max_restarts_number          int2                        NOT NULL DEFAULT 100,
+    time_between_restarts        int2                        NOT NULL DEFAULT 1,
+    last_restarted_at            timestamp without time zone,
+    last_error                   jsonb,
+    last_error_occurred_at       timestamp without time zone,
+    chunk_query_interval         int2                        NOT NULL DEFAULT 5,
+    last_chunk_fed_at            timestamp without time zone NOT NULL DEFAULT to_timestamp(0),
+    last_chunk_greatest_position bigint,
```

Comment:
It is used to determine the position to fetch events from. 

During the events processing feeder runner does not fetch events one by one and it does not wait for the Subscription's runner to process events in the queue. Thus, to prevent fetching duplicated events(in case if we would like to refer on Subscription's position instead) - we memorize the position of the last event in the last chunk being fed to the Subscription's runner

Reply:
dont quite understand this. so this is not an absolute position, but relative inside a chunk?

Reply:
Yes, exactly. I can explain during the call